platforms/ide/ide/src/main/java/org/gradle/plugins/ide/internal/tooling/ResilientGradleBuildBuilder.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2013 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.plugins.ide.internal.tooling;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.initialization.ProjectDescriptor;
+import org.gradle.api.internal.GradleInternal;
+import org.gradle.api.internal.SettingsInternal;
+import org.gradle.internal.build.BuildState;
+import org.gradle.internal.build.BuildStateRegistry;
+import org.gradle.internal.composite.BuildIncludeListener;
+import org.gradle.internal.problems.failure.Failure;
+import org.gradle.plugins.ide.internal.tooling.model.BasicGradleProject;
+import org.gradle.plugins.ide.internal.tooling.model.DefaultGradleBuild;
+import org.gradle.plugins.ide.internal.tooling.model.DefaultResilientGradleBuild;
+import org.gradle.tooling.internal.gradle.DefaultProjectIdentifier;
+import org.jspecify.annotations.NullMarked;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@NullMarked
+public class ResilientGradleBuildBuilder extends GradleBuildBuilder {
```

Comment:
Why use inheritance here? Wouldn't it be clearer if we were to use composition? Inheritance is weird on a conceptual level too. Why would the resilient builder extend the "legacy" one? It feels much more natural to just share some code.

Reply:
inheritence because I think reuse with composition is a lot more cumbersome, since part of it is just overriding behaviour in some methods and keep the rest intact.